googletest/include/gtest/gtest-printers.h

```diff
@@ -423,9 +423,9 @@ void DefaultPrintTo(IsNotContainer /* dummy */,
       // void*.  However, we cannot cast it to const void* directly,
       // even using reinterpret_cast, as earlier versions of gcc
       // (e.g. 3.4.5) cannot compile the cast when p is a function
```

Comment:
Alternative idea to fix your warning.
Let's just get rid of the gcc 3.4.5 support here, and use `*os << reinterpret_cast<const void*>(p);`.